### PR TITLE
Migrate automatic python resource list to Homebrew/core

### PR DESCRIPTION
**Pull request by Rylan12**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This PR migrates the `AUTOMATIC_RESOURCE_UPDATE_BLOCKLIST` list to Homebrew/core in conjunction with https://github.com/Homebrew/brew/pull/9185.

This places control over the automatic resource bumps for Python formulae in this repo. Here is and example `automatic_resource_update_blocklist.json` file:

```json
{
  "foo": false,
  "bar": "abc",
  "baz": {
    "package_name": "abc",
    "extra_packages": ["extra1", "extra2"],
    "exclude_packages": ["exclude1", "exclude2"]
  }
}
```

There are three type of declarations:

1. `"foo": false` means that the `foo` formula will be prohibited from having its resources automatically bumped (similar to the existing blocklist)
1. `"bar": "bar[abc]"` means that the `bar` formulae will retrieve the resources from the `abc` PyPI package _instead of_ the `bar` package (this can also accept packages like `bar[abc]` or `bar[abc,def]`). This corresponds to the `--package-name` flag.
1. `"baz": {` is for more complex situations:
    - `"package_name": "abc"` is equivalent to number 2 above. It will use `abc` instead of `baz` for the package name.
        - This corresponds to the `--package-name` flag
    - `"extra_packages": ["extra1", "extra2"]` means that the resources and requirements for the `extra1` and `extra2` packages will be included _along with_ those from the main package. This is a way to add additional resource blocks that won't be picked up normally (you can also specify specific versions by using `extra1==1.2.3` in addition to packages like `abc[def]`).
        - This corresponds to the `--extra-packages` flag
    - `"exclude_packages": ["exclude1", "exclude2"]` means that the resources for the `exclude1` and `exclude2` packages will _not_ be in included. This is useful if `pipgrip` picks up unnecessary dependencies.
        - This corresponds to the `--exclude-packages ` flag

### CODE REVIEW

**Review by SMillerDev — DISMISSED** [review]

**Comment by Rylan12** [comment]
Putting this on hold until the migration in https://github.com/Homebrew/brew/pull/9178 is finished.

**Comment by Rylan12** [comment]
I would appreciate it if someone (maybe @dtrodrigues if you've got some time) could double-check my mappings here.

A few brief notes:

- `ansible` and `ansible@2.8` are complex and should be done manually (hence being set to `false`)
- `molecule` seems like it requires the `molecule-vagrant` package which is not included in `molecule`'s dependencies. However, specifying `molecule-vagrant` to `--extra-packages` fails because `molecule-vagrant` depends on `molecule==3.2.0a0` which causes a version conflict. For now, I've set `molecule` to `false`, disabling auto updates. Once this and the Homebrew/brew PR are merged, I'll update the comments in the `molecule` formula with these instructions for updating:
    - ``First, run `brew update-formula-resources molecule --extra-packages docker-py,python-vagrant` (this will remove the `molecule-vagrant` resource)``
    - ``Then, manually re-add the `molecule-vagrant` resource using the link and checksum found at `https://pypi.org/project/molecule-vagrant/#files` ``

File: formula_lists/pypi_automatic_resource_update_list.json
Comment (dtrodrigues) [inline formula_lists/pypi_automatic_resource_update_list.json:9]: `ipykernel` can technically be an "instead of" vs an "extra" since `ipykernel` depends on `ipython`, but its version numbering doesn't match that of `ipython`, so it can't  be specified in the same way as `diffocsope`. The end result is the same and it's such a small optimization that it's probably not worth adding extra logic in for.

**Review by dtrodrigues — DISMISSED** [review]
Great stuff!

File: formula_lists/pypi_automatic_resource_update_list.json
Comment (Rylan12) [reply formula_lists/pypi_automatic_resource_update_list.json:9]: Yeah, I looked at that but specifying `ipykernal` as the package name instead of `ipython` doesn't totally work for a few reasons:
1. The version issue you mentioned
1. This would treat the formula as if it were an `ipykernal` formula meaning that it would add `ipython` as a resource and wouldn't actually include `ipykernal` as a resource.

Once all this stuff is merged I'll make sure to go through these formulae and update the comments describing the resource issues.

Thanks for the review! I'll get this merged once everything in https://github.com/Homebrew/brew/pull/9185 is resolved.

**Comment by Rylan12** [comment]
Ah, one other thing I forgot to mention: `pipgrip dxpy` no longer returned the `gnureadline` dependency (which was the reason it was on the blocklist) so I removed it. I wouldn't mind being double-checked that it works correctly, now.

**Comment by Rylan12** [comment]
(No changes, just changes the filename per the brew PR)

**Review by SMillerDev — DISMISSED** [review]

**Comment by Rylan12** [comment]
I added `xdot` as mentioned in https://github.com/Homebrew/homebrew-core/pull/65354.

I'm not sure why BrewTestBot keeps adding and removing `automerge-skip`... No big deal though. I will merge this manually when https://github.com/Homebrew/brew/pull/9185 is ready, so no need to continue to approve each time I update (although maintainers are always welcome to approve/give suggestions).

**Pull request merged**
